lib/puppet/type/cisco_bridge_domain.rb

```diff
@@ -0,0 +1,110 @@
+# Manages a Cisco Bridge Domain.
+#
+# March 2016
+#
+# Copyright (c) 2016 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+Puppet::Type.newtype(:cisco_bridge_domain) do
+  @doc = "Manages a Cisco Bridge Domain (BD).
+
+  cisco_bridge_domain {\"<bd>\":
+    ..attributes..
+  }
+
+  <bd> is the id of the bridge domain.
+
+  Example:
+    cisco_bridge_domain {\"1000\":
+      ensure          => present,
+      bd_name         => 'red',
+      fabric_control  => 'false',
+      shutdown        => 'false',
+    }
+  "
+
+  ###################
+  # Resource Naming #
+  ###################
+
+  # Parse out the title to fill in the attributes in these
+  # patterns. These attributes can be overwritten later.
+  def self.title_patterns
+    identity = ->(x) { x }
+    patterns = []
+
+    # Below pattern matches both parts of the full composite name.
+    patterns << [
+      /^(\d+)$/,
+      [
+        [:bd, identity]
+      ],
+    ]
+    patterns
+  end
+
+  newparam(:bd, namevar: true) do
+    desc 'ID of the Bridge Domain. Valid values are integer.'
+
+    validate do |value|
+      range = *(2..4096)
+      internal = *(3968..4096)
+      valid_ids = range - internal
```

Comment:
Why break this into `range` and `internal` instead of just `valid_ids = *(2..3967)` ? For documentation?